### ISSUE TITLE
Unable Login to CRM when Import solution to a NEWLY created environment 

### ISSUE THREAD

**Issue by lheng18** [WaelHamze/dyn365-ce-vsts-tasks#99]
Hi @WaelHamze, I encountered an strange issue. When using Import solution with correct connection string, I got error Unable Login to CRM to a newly created Power/365 online environment created using API in the first day or so but the issue is gone after a few days. Here is the error from the build task in the pipeline.
##[section]Starting: MSCRM Import Solution
==============================================================================
Task         : MSCRM Import Solution
Description  : Import a Dynamics CRM Solution package
Version      : 11.0.4
Author       : Wael Hamze
Help         : This task invokes ImportSolutionRequest in Async mode
==============================================================================
##[warning]Unable to Login to Dynamics CRM
Unable to Login to Dynamics CRM
##[warning]Unable to Login to Dynamics CRM
##[warning]Unable to Login to Dynamics CRM
Unable to Login to Dynamics CRM
##[warning]Unable to Login to Dynamics CRM
##[warning]Unable to Login to Dynamics CRM
Unable to Login to Dynamics CRM
##[warning]Unable to Login to Dynamics CRM
##[error]Couldn't connect to CRM instance after 3 attempts: Unable to Login to Dynamics CRM
Unable to Login to Dynamics CRM
##[section]Finishing: MSCRM Import Solution

 I suspect something in the newly created CRM was not ready yet until a few days later, maybe like user sync still being in progress, but the user/service account that was used to connect CRM in connection string has been already existed with System Admin role in the CRM actually I am able login CRM using the user's credentials. Besides, I am able to connect the new environment using Microsoft Plugin Registration tool with the user but not when using the build task in the pipeline. Further more, I encountered the same issue when connecting CRM in XRMToolbox.
here are the error from XRMToolbox. it may be help to troubleshoot the issue. Maybe the client xrm.tool sdk assembly has a problem and MS Plugin Registration tool doesn't use the same assembly?
======================================================================================================================

Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Verbose	16	7/16/2019 6:04:31 AM  Initialize CRM connection Started - AuthType: Office365
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Information	8	7/16/2019 6:04:31 AM  Using User Specified Server 
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Information	8	7/16/2019 6:04:31 AM  DiscoverOrganizations - Initializing Discovery Server Object with https://disco.crm.dynamics.com/XRMServices/2011/Discovery.svc
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Verbose	16	7/16/2019 6:04:31 AM  DiscoverOrganizations - attempting to connect to CRM server @ https://disco.crm.dynamics.com/XRMServices/2011/Discovery.svc
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Verbose	16	7/16/2019 6:04:31 AM  DiscoverOrganizations - created CRM server proxy configuration for https://disco.crm.dynamics.com/XRMServices/2011/Discovery.svc - duration: 00:00:00.1986210
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Verbose	16	7/16/2019 6:04:31 AM  DiscoverOrganizations - proxy requiring authentication type : OnlineFederation 
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Verbose	16	7/16/2019 6:04:31 AM  DiscoverOrganizations - Authenticated via OnlineFederation. Auth Elapsed:00:00:00.1975214
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Verbose	16	7/16/2019 6:04:31 AM  DiscoverOrganizations - service proxy created - total create duration: 00:00:00.4005431
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Information	8	7/16/2019 6:04:32 AM  DiscoverOrganizations - Discovery Server Get Orgs Call Complete - Elapsed:00:00:01.1354548
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Information	8	7/16/2019 6:04:32 AM  Found 559 Org(s)
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Information	8	7/16/2019 6:04:32 AM  Looking for Organization = org8a88a307 in the results from CRM's Discovery server list.
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Error	2	7/16/2019 6:04:32 AM  Unable to Login to Dynamics CRM
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Error	2	7/16/2019 6:04:32 AM  Unable to Login to Dynamics CRM
Source	: Not Provided
Method	: Not Provided
Date	: 7/16/2019
Time	: 6:04:32 AM
Error	: Unable to Login to Dynamics CRM
Stack Trace	: Not Provided
======================================================================================================================

Your help is very appreciated!

**Comment by WaelHamze** [comment]
@lheng18 

Does this happen regularly?

I would create the instance using same user as importing solution and ensure this is a dynamics 365 admin.

If needed some delay between creating instance and importing solutions to allow user sync etc to complete.

**Comment by lheng18** [comment]
Yes, every single time. The service account that is used to login is already added as system admin using power platform API Assign to Admin role in flow right after the environment is created before creating CDS.I can login into the url from browser after the CDS is created using API using the user. But can’t connect from your tool and xrm tool box and even power shell crm-connect in power data or the old connecting way in Xrm module until 2 days later using the user.I can use Microsoft newly released build tool Import Solution build task fine. But their tool needs hard coded service connection not supporting connection string. I can use an admin account that was used to create the environment initially through api to connect to the enviroment using API fine. just the service account or any users that were added in the above way not working until a few days. thanks.

Sent from my iPhone

On Aug 8, 2019, at 1:16 PM, WaelHamze <notifications@github.com<mailto:notifications@github.com>> wrote:


@lheng18<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Flheng18&data=02%7C01%7C%7C9f1cb2a894a244852c5d08d71c3d500b%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637008921964436490&sdata=vrCod2QcLl2xNLmeLg%2FhTQbyjK0ZJhpPu%2B418nzQA38%3D&reserved=0>

Does this happen regularly?

I would create the instance using same user as importing solution and ensure this is a dynamics 365 admin.

If needed some delay between creating instance and importing solutions to allow user sync etc to complete.

**Comment by WaelHamze** [comment]
@lheng18 did you raise this with MS?

If you login to the instance an you see the user and it is assigned system admin?

**Comment by lheng18** [comment]
Yes. raised ticket with them. Yes, I saw the user is system admin after I login in. I can even login with the users credentials  to the environment in browser.

Sent from my iPhone

On Aug 8, 2019, at 2:18 PM, WaelHamze <notifications@github.com<mailto:notifications@github.com>> wrote:


@lheng18<https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Flheng18&data=02%7C01%7C%7Cc3e5ea37e6b44fb76e4a08d71c45f86c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637008959147970835&sdata=ojaxzmib9uhEsD13Gxl0g8TgkgAzUQxrhRuIL02LQXE%3D&reserved=0> did you raise this with MS?

If you login to the instance an you see the user and it is assigned system admin?

**Comment by WaelHamze** [comment]
@lheng18 What was the outcome of the ticket?
